internal/infrastructure/kubernetes/proxy/resource.go

```diff
@@ -135,7 +131,7 @@ func expectedProxyContainers(infra *ir.ProxyInfra,
 			ImagePullPolicy:          corev1.PullIfNotPresent,
 			Command:                  []string{"envoy"},
 			Args:                     args,
-			Env:                      expectedContainerEnv(containerSpec, gatewayNamespaceMode),
+			Env:                      expectedContainerEnv(containerSpec, egNamespace),
```

Comment:
Nit, non-blocking: Can we use a consistent name for `controllerNamespace/egNamespace`. Two names for the same concept might confuse future readers.

I slightly prefer `EGNamespace` as it feels a bit more straightforward.

Reply:
API calls it ControllerNamespace

Reply:
ControllerNamespace is fine.